agent/internal/redactor/redacting_sender.go

```diff
@@ -28,6 +30,11 @@ type RedactionOptions struct {
 	// Redacted specifies the names of expansions to redact the values for.
 	// [globals.ExpansionsToRedact] are always redacted.
 	Redacted []string
+	// AdditionalRedactions specifies an additional set of strings that are not
+	// expansions that should be redacted from the logs (e.g. agent-internal
+	// secrets). All key-values in AdditionalRedactions are assumed to be
+	// sensitive.
+	AdditionalRedactions *util.DynamicExpansions
```

Comment:
Can we also name this `InternalRedactions`, since its only purpose would be for internal secrets?